### PR TITLE
[FLINK-34467] bump flink version to 1.20.0

### PR DESCRIPTION
**Pull request by HuangZhenQiu**
Bump flink version to 1.20.0 for preparing lineage integration.

### CODE REVIEW

**Comment by leonardBang** [comment]
@lvyanquan Would you like to review this PR when you have time?

**Comment by HuangZhenQiu** [comment]
The build is successful in my personal CI

https://github.com/HuangZhenQiu/flink-connector-jdbc/actions/runs/11328373650

**Comment by HuangZhenQiu** [comment]
@flinkbot run

**Comment by lvyanquan** [comment]
I've read your discussions in Kafka connector community, as we already release a 3.2 version to support flink 1.19, and refer to [AHeise's comment](https://github.com/apache/flink-connector-kafka/pull/111#issuecomment-2309462158), we can add bugfixs to 3.2.1 if necessary. 
So this version bumping is acceptable to me, and we need a follow-up PR to set the build version to 3.3-SNAPSHOT to avoid occupying the minor version 3.2.1.

File: .github/workflows/push_pr.yml [hunk 1]
```diff
@@ -28,10 +28,10 @@ jobs:
   compile_and_test:
     strategy:
       matrix:
-        flink: [ 1.19-SNAPSHOT, 1.20-SNAPSHOT ]
+        flink: [ 1.19.1 ]
```
[the remark this answers is not in the document]
Comment (lvyanquan) [reply .github/workflows/push_pr.yml:31]: Yeah, so we could skip check the 1.19 version in pr-push too.

File: .github/workflows/push_pr.yml [hunk 2]
```diff
@@ -28,12 +28,8 @@ jobs:
...
     strategy:
       matrix:
-        flink: [ 1.19-SNAPSHOT, 1.20-SNAPSHOT ]
-        jdk: [ '8, 11, 17, 21' ]
-        include:
-          - flink: 1.18-SNAPSHOT
-            jdk: '8, 11, 17'
-
+        flink: [1.20.0]
+        jdk: ['8, 11, 17']
```
Comment (lvyanquan) [inline .github/workflows/push_pr.yml:32]: jdk21 was missed.

**Review by lvyanquan — APPROVED** [review]
LGTM.

**Review by leonardBang — APPROVED** [review]
+1 from my side, thanks @HuangZhenQiu @lvyanquan for the nice work

**Pull request merged**